### DISCUSSION TITLE
Auto-detect if routing can be skipped

### DISCUSSION THREAD

**Discussion by behrisch** [PedestrianDynamics/jupedsim#1497] [Report Issues / Request Features]
The current routing engine can not only give the shortest path but also a guarantee that the second shortest path is at least _x_ meters longer. If we assume that the underlying geometry has not changed (and of course the target is still the same), we can use this value to determine if routing is actually necessary.

The idea is to store the previous position of the agent and the gap to the second longest walk _x_. We can then calculate how much the distance to the next point in the current shortest path decreased (let's say _s_) and we know the absolute distance traveled since the last routing (let's say _d_). So if _s + x > d_ we don't need to do a routing.

**Comment by Ozaq** [comment]
Lets discuss this in #1392
